sycl/plugins/cuda/pi_cuda.cpp

```diff
@@ -2608,6 +2608,10 @@ pi_result cuda_piEnqueueKernelLaunch(
   assert(work_dim > 0);
   assert(work_dim < 4);
 
+  if (*global_work_size == 0) {
+    return PI_SUCCESS;
+  }
```

Comment:
Doing an early exit here means that we do not create an event. I fear that could cause unexpected problems. Same goes for HIP.

Reply:
@steffenlarsen I agree with you on this regarding the event. 
If I  move the condition (**if (*global_work_size =! 0 )**) to guard the call to  **PI_CHECK_ERROR(cuLaunchKernel(** it will preserve the functionality related to event handling. Do you have any concern doing that?
I have not tried the L0, but the opencl:cpu and esimd (and host in an earlier version of DP++) backend are tried and work without any modifications.

Reply:
Either that or you could copy the event creation to here, something like:

```c++
if (event) {
  std::unique_ptr<_pi_event> retImplEv{nullptr};
  retImplEv = std::unique_ptr<_pi_event>(_pi_event::make_native(
      PI_COMMAND_TYPE_NDRANGE_KERNEL, command_queue));
  retImplEv->start();
  retImplEv->record();
  *event = retImplEv.release();
}
```

Reply:
I'm not sure it will be a correct implementation. A call with `NDRange == 0` while should do nothing in terms of running the kernel, the event produced should be as if we run a kernel:
1. Completion of such an event should guarantee completion of events passed in `event_wait_list`
2. The event can be used in a `event_wait_list` of a subsequent enqueue.

So, I think that the more correct implementation would be just calling `cuda_piEnqueueEventsWaitWithBarrier` if `NDRange == 0`.

Reply:
That is a good point. Now that there are multiple queues, recording an event is not enough to act like a barrier.